src/lib/proof-system/zkprogram.ts

```diff
@@ -351,61 +376,49 @@ function ZkProgram<
     }
   }
 
-  function toProver<K extends keyof Methods & string>(
+  type RegularProver<K extends MethodKey> = (
```

Comment:
All the other types are at the bottom of this file.  Perhaps we ought to break out some of the type declarations since this file is so massive now.

What makes this type `Regular`?  Is it in opposition to `RecursiveProver`?  The new types could use documentary comments I think.

Reply:
this type is not at the bottom of the file since it depends on the generic type params!

it's a `RegularProver` as opposed to the normal `Prover`, because it expects a public input argument, even if the public input type is `undefined`. In the final `Prover`s that are exposed on the program, we add some additional wrapping so that users don't have to pass in an `undefined` input, which mimics the way the methods themselves are also defined.

regular = method signature doesn't depend on the runtime publicInputType

for the internal logic, the "regular" prover signature is much easier to deal with, that's why I separated out the wrapping to remove the public input argument into a different step, so that I can use the regular prover for the new _recursive_ prover

Reply:
> this type is not at the bottom of the file since it depends on the generic type params!

Ah right!

> it's a RegularProver as opposed to the normal Prover, because it expects a public input argument, even if the public input type is undefined

Could you add a comment to that effect in the code so in the future a reader will understand that at a glance?